templates/cicd/pipeline_cfn.yml

```
@@ -3,13 +3,13 @@
 AWSTemplateFormatVersion: '2010-09-09'
 Description: CodePipeline for {{$.AppName}}
 Resources:
-  {{- if isCodeStarConnection .Source}}
+  {{- if isCodeStarConnection .Source}}{{if eq .Source.ConnectionARN ""}}
```

nit: can we move the actions to separate lines to increase readability:
```suggestion
  {{- if isCodeStarConnection .Source}}
  {{- if eq .Source.ConnectionARN ""}}
  {{- end }}
  {{- end}}
```
The minus sign `{{-` is go template magic to trim all trailing white space from the immediately preceding text.